### PR TITLE
Implement typing

### LINKED ISSUE #119: Transactions
Hello!

Looks like no type definitions for transactions..
Am I wrong?

### LINKED ISSUE #142: Add Typescript support
TypeScript support is currently maintained in [DefinitelyTyped/DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped). Consider providing TypeScript definitions in this repository.

I probably won't have time to do this anytime soon, so if someone wants to take this on feel free to open a PR (or submit fixes to DefinitelyTyped).

### PR DESCRIPTION
**Pull request by UNIDY2002**
Fix #142
Fix #119

Advantages of my implementation:
- Up to date with the latest `node-consul`
- Try my best to type the API results as well
- No modification to the JS code

Drawbacks of my implementation:
- Haven't tested thoroughly

Typing is hard to test, though.

Nevertheless, since typing mainly serves as an auxiliary facility, I think the best way of testing is to try it out in existing projects, and I am willing to fix things if any incorrectness is reported.

Looking forward to your review.

### CODE REVIEW

**Comment by UNIDY2002** [comment]
Also fix #119

**Comment by silas** [comment]
This is awesome and thanks for doing the work!

I agree that we should probably just start using this and fix stuff as issues arise. I've been sick for the last couple of days and now I'm catching up on work stuff, but I'll try to get this reviewed/merged in the next week.

File: package.json
Comment (wregis) [inline package.json:1]: Shouldn't there be a `types` pointing to `lib/index.d.ts`?

Comment (UNIDY2002) [reply package.json:1]: You are right. Thank you for reminding me!

**Comment by silas** [comment]
Thanks again @UNIDY2002! I just pushed consul@2.0.0-next.1 (`consul@next`) with these changes.

I figure we can just iterate on a pre-release version for a bit until we're comfortable with the types.

**Comment by UNIDY2002** [comment]
I agree, thanks!

**Pull request merged**